plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -0,0 +1,176 @@
+      .map(r => r.targetRef) ?? [];
+
+  return Array.from(new Set([stringifyEntityRef(entity), ...membershipRefs]));
+}
+
+/**
+ * @internal
+ */
+export class CatalogAuthResolverContext implements AuthResolverContext {
+  static create(options: {
+    logger: Logger;
+    catalogApi: CatalogApi;
+    tokenIssuer: TokenIssuer;
+    tokenManager: TokenManager;
+  }): CatalogAuthResolverContext {
+    const catalogIdentityClient = new CatalogIdentityClient({
+      catalogApi: options.catalogApi,
+      tokenManager: options.tokenManager,
+    });
+    return new CatalogAuthResolverContext(
+      options.logger,
+      options.tokenIssuer,
+      catalogIdentityClient,
+      options.catalogApi,
+      options.tokenManager,
+    );
+  }
+
+  private constructor(
+    public readonly logger: Logger,
+    public readonly tokenIssuer: TokenIssuer,
+    public readonly catalogIdentityClient: CatalogIdentityClient,
+    private readonly catalogApi: CatalogApi,
+    private readonly tokenManager: TokenManager,
+  ) {}
+
+  async issueToken(params: TokenParams) {
+    const token = await this.tokenIssuer.issueToken(params);
+    return { token };
+  }
+
+  async findCatalogUser(query: AuthResolverCatalogUserQuery) {
+    let result: Entity[] | Entity | undefined = undefined;
+    const { token } = await this.tokenManager.getToken();
+
+    if ('entityRef' in query) {
+      const entityRef = parseEntityRef(query.entityRef, {
+        defaultKind: 'user',
+        defaultNamespace: DEFAULT_NAMESPACE,
+      });
+      result = await this.catalogApi.getEntityByRef(entityRef, { token });
+    } else if ('annotations' in query) {
+      const filter: Record<string, string> = {
+        kind: 'user',
+      };
+      for (const [key, value] of Object.entries(query.annotations)) {
+        filter[`metadata.annotations.${key}`] = value;
+      }
+      const res = await this.catalogApi.getEntities({ filter }, { token });
+      result = res.items;
+    } else if ('filter' in query) {
+      const res = await this.catalogApi.getEntities(
+        { filter: query.filter },
+        { token },
+      );
+      result = res.items;
+    } else {
+      throw new InputError('Invalid user lookup query');
+    }
+
+    if (Array.isArray(result)) {
+      if (result.length > 1) {
+        throw new ConflictError('User lookup resulted in multiple matches');
+      }
+      result = result[0];
+    }
+    if (!result) {
+      throw new NotFoundError('User not found');
+    }
+
+    return { entity: result };
+  }
+
+  async signInWithCatalogUser(query: AuthResolverCatalogUserQuery) {
+    const { entity } = await this.findCatalogUser(query);
+    const ownershipRefs = getDefaultOwnershipEntityRefs(entity);
+
+    const token = await this.tokenIssuer.issueToken({
+      claims: {
+        sub: stringifyEntityRef(entity),
+        ent: ownershipRefs,
+      },
+    });
+    return { token };
+  }
+  /*
+  async expandCatalogOwnership(query: { entityRefs: string[] }) {
```

Comment:
I migrated this but it ended up feeling redundant. Not sure when this would really be used in the end tbh, especially now with the new `getDefaultOwnershipEntityRefs` most of what this does is just dealing with the fact that we support multiple entities at once and the extra checks that it requires.